File: src/main/java/org/railwaystations/rsapi/core/services/CountryService.java
package org.railwaystations.rsapi.core.services;

import org.railwaystations.rsapi.adapter.out.db.CountryDao;
import org.railwaystations.rsapi.core.model.Country;
import org.railwaystations.rsapi.core.ports.in.ListCountriesUseCase;
import org.springframework.stereotype.Service;

import java.util.Collection;

@Service
public class CountryService implements ListCountriesUseCase {

    private final CountryDao countryDao;

    public CountryService(CountryDao countryDao) {
        this.countryDao = countryDao;
    }

    @Override
    public Collection<Country> list(Boolean onlyActive) {
        return countryDao.list(onlyActive == null || onlyActive);
    }

}

File: service/src/main/kotlin/org/railwaystations/rsapi/core/services/CountryService.kt
package org.railwaystations.rsapi.core.services

import org.railwaystations.rsapi.adapter.db.CountryDao
import org.railwaystations.rsapi.core.model.Country
import org.railwaystations.rsapi.core.ports.ListCountriesUseCase
import org.springframework.stereotype.Service

@Service
class CountryService(private val countryDao: CountryDao) : ListCountriesUseCase {
    override fun list(onlyActive: Boolean?): Collection<Country> {
        return countryDao.list(onlyActive == null || onlyActive)
    }
}
